src/main/perl/Kerberos.pm

```diff
@@ -314,6 +319,18 @@ sub get_context
         return if(! defined($name));
     }
 
+    my $cred;
+
+    if ($opts{usecred}) {
+        $self->verbose("trying to get credential");
+        $cred = $self->get_cred(name => $name);
+        # Already reports error
+        return if ! $cred;
+    } else {
+        $self->verbose("No checking for credential");
```

Comment:
i'm not sure. the context mechanism is used to get the initial tokens, but without passing a credential instance, the principal passed via the name instance is used in a sort of best-effort way.